### ISSUE TITLE
Library fails on Firefox private mode

### ISSUE THREAD

**Issue by sebaherrera** [cyrilletuzi/angular-async-local-storage#184]
Hi. Fallback to in-memory storage isn't working on Firefox private mode. Instead, I get "TypeError: t is undefined".

I was able to reproduce it and make a StackBlitz: https://stackblitz.com/edit/angular-sxmxwj. The only difference is that the error says "TypeError: "database is undefined".

Is there again an issue like https://github.com/cyrilletuzi/angular-async-local-storage/issues/26?

Stack trace caught on production with same environment, except for zone@0.9.1 and typescript@3.5.3:

```
File webpack:///node_modules/@ngx-pwa/local-storage/fesm2015/ngx-pwa-local-storage.js.pre-build-optimizer.js line 628 col 17 in transaction/<
transaction = database.transaction([this.storeName], mode);
File webpack:///node_modules/rxjs/_esm2015/internal/operators/mergeMap.js.pre-build-optimizer.js line 46 col 27 in _tryNext
result = this.project(value, index);
File webpack:///node_modules/rxjs/_esm2015/internal/operators/mergeMap.js.pre-build-optimizer.js line 36 col 18 in _next
this._tryNext(value);
File webpack:///node_modules/rxjs/_esm2015/internal/Subscriber.js.pre-build-optimizer.js line 49 col 18 in next
this._next(value);
File webpack:///node_modules/rxjs/_esm2015/internal/ReplaySubject.js.pre-build-optimizer.js line 57 col 28 in _subscribe
subscriber.next(_events[i]);
File webpack:///node_modules/rxjs/_esm2015/internal/Observable.js.pre-build-optimizer.js line 42 col 25 in _trySubscribe
return this._subscribe(sink);
File webpack:///node_modules/rxjs/_esm2015/internal/Subject.js.pre-build-optimizer.js line 81 col 26 in _trySubscribe
return super._trySubscribe(subscriber);
File webpack:///node_modules/rxjs/_esm2015/internal/Observable.js.pre-build-optimizer.js line 28 col 22 in subscribe
this._trySubscribe(sink));
File webpack:///node_modules/rxjs/_esm2015/internal/operators/mergeMap.js.pre-build-optimizer.js line 21 col 23 in call
return source.subscribe(new MergeMapSubscriber(observer, this.project, this.concurrent));
File webpack:///node_modules/rxjs/_esm2015/internal/Observable.js.pre-build-optimizer.js line 23 col 31 in subscribe
sink.add(operator.call(sink, this.source));
File webpack:///node_modules/rxjs/_esm2015/internal/operators/take.js.pre-build-optimizer.js line 22 col 23 in call
return source.subscribe(new TakeSubscriber(subscriber, this.total));
File webpack:///node_modules/rxjs/_esm2015/internal/Observable.js.pre-build-optimizer.js line 23 col 31 in subscribe
sink.add(operator.call(sink, this.source));
File webpack:///node_modules/rxjs/_esm2015/internal/operators/throwIfEmpty.js.pre-build-optimizer.js line 13 col 23 in call
return source.subscribe(new ThrowIfEmptySubscriber(subscriber, this.errorFactory));
File webpack:///node_modules/rxjs/_esm2015/internal/Observable.js.pre-build-optimizer.js line 23 col 31 in subscribe
sink.add(operator.call(sink, this.source));
File webpack:///node_modules/rxjs/_esm2015/internal/operators/mergeMap.js.pre-build-optimizer.js line 21 col 23 in call
return source.subscribe(new MergeMapSubscriber(observer, this.project, this.concurrent));
File webpack:///node_modules/rxjs/_esm2015/internal/Observable.js.pre-build-optimizer.js line 23 col 31 in subscribe
sink.add(operator.call(sink, this.source));
File webpack:///node_modules/rxjs/_esm2015/internal/operators/catchError.js.pre-build-optimizer.js line 16 col 23 in call
return source.subscribe(new CatchSubscriber(subscriber, this.selector, this.caught));
File webpack:///node_modules/rxjs/_esm2015/internal/Observable.js.pre-build-optimizer.js line 23 col 31 in subscribe
sink.add(operator.call(sink, this.source));
File webpack:///node_modules/rxjs/_esm2015/internal/operators/mergeMap.js.pre-build-optimizer.js line 21 col 23 in call
return source.subscribe(new MergeMapSubscriber(observer, this.project, this.concurrent));
File webpack:///node_modules/rxjs/_esm2015/internal/Observable.js.pre-build-optimizer.js line 23 col 31 in subscribe
sink.add(operator.call(sink, this.source));
```

**Comment by cyrilletuzi** [comment]
Thanks for your report.

Seems to be a regression introduced by 8.2 (ie. #169). I'll fix this as soon as possible.

**Comment by cyrilletuzi** [comment]
Fix released in 8.2.2

**Comment by sebaherrera** [comment]
Thanks! BTW, error callbacks are working right so the only remaining issue may be https://github.com/cyrilletuzi/angular-async-local-storage/issues/95#issuecomment-496193405. Great job!
